connectors/sources/confluence.py

```diff
@@ -1231,3 +1332,6 @@ async def get_docs(self, filtering=None):
             async for item in self._consumer():
                 yield item
             await self.fetchers.join()
+
+        if self.confluence_client.api_total_count > 0:
```

Comment:
should this be an even higher number? Like IDK if we really should call it a failure if there was 1 error out of 10 total API calls. That doesn't feel statistically significant. 

Or maybe we should just make this number another constant that we can let folks edit if they want to adjust the minimum threshold?

Reply:
> should this be an even higher number? Like IDK if we really should call it a failure if there was 1 error out of 10 total API calls. That doesn't feel statistically significant.


Adding this conditional check to just serves as a protective measure to prevent division by zero errors in the calculation process.

Reply:
> Or maybe we should just make this number another constant that we can let folks edit if they want to adjust the minimum threshold?

Already assigning the constant `API_FAILURE_THRESHOLD` a predefined value, folks have the option to easily adjust this threshold by editing the constant if needed to tailor the failure criteria to their requirements.

Reply:
Yes, but `N%` is not always significant. For example, if >10% of 1000000 requests fail, that's a LOT of failures. That probably means there's a significant issue. But if >10% of 2 requests fail, that might just be a fluke. You can address this by having two thresholds - one for a minimum sample size, and one for a percentage.

In Workplace Search, we had those two as well as a two others, (a window size, and a window percentage). So that you could fail if 30% of a window of 100 failed, or if 20% overall failed, but only if you had more than 1000 total. I don't think we need to get that complex, but I'm trying to illustrate the nuance that we've found important in the past.